### ISSUE TITLE
System.TypeLoadException

### ISSUE THREAD

**Issue by metheofanis** [ErikEJ/EntityFramework.SqlServerCompact#584]
I've return to an old project.
While build is successful, I can not run it any more. I get a run time exception like this:

> Could not load type 'System.Collections.Generic.IAsyncEnumerable`1' from assembly 'System.Interactive.Async, Version=5.0.0.0, Culture=neutral, PublicKeyToken=94bc3704cddfc263'.

What do i miss?

**Comment by ErikEJ** [comment]
Pls share your .csproj??

**Comment by metheofanis** [comment]
Here it is.

[Myriobiblos.AlfaExtend.zip](https://github.com/ErikEJ/EntityFramework.SqlServerCompact/files/6056414/Myriobiblos.AlfaExtend.zip)

**Comment by ErikEJ** [comment]
I guess it is the version 5 of many Microsoft.Extension packages that does not paly well with the old .NET Core 2.2 package ( like Microsoft.Bcl.AsyncInterfaces.5.0.0)

**Comment by ErikEJ** [comment]
Maybe this is related: https://github.com/dotnet/efcore/issues/18124

**Comment by metheofanis** [comment]
I'm sorry I'm a novice programmer. 
So, what is the correct versions of .net and entityframework? I've spend a lot of time downgrading and upgrading everything, with out success. Shouldn't this nuget packet pull the correct versions automatically?

**Comment by ErikEJ** [comment]
I think you may have other components in use that depend on newer versions.

Start simple with a console app, without any other dependencies than the ef core provider.

**Comment by metheofanis** [comment]
Currently, I'm targeting .net 4.7.2. 
I've removed **all** nuget packages from my project, and all code related to them. I've "cleaned" the solution, the local nuget packages, deleted obj and bin folders. 
Then I've installed this packet. It automatically installed a bunch of other dependencies. The project builds successfully. At run time I get the same error. The first time the program "hits" the database I get it.
Again I've spent a lot of time upgrading and downgrading everything, targeting different .net versions. No luck.
The good news is it is working on a console app, targeting .netcore30.
Any suggestions for the .net 4.7.2 ?
